src/pluginLoader.js

```diff
@@ -0,0 +1,660 @@
+          if (messageSplit[0].search(lookFor) === 0) {
+            var msg = messageSplit[0].slice(lookFor.length);
+            window['JSConsole'].info("msg: ", msg);
+            try {
+              var errorObject = JSON.parse(msg);
+              var error = new Error();
+              error.message = errorObject['error'];
+              error.stack = errorObject['stacktrace'].replace("\\t", "&nbsp;&nbsp").replace("\\n", "\n");
+              messages = [error];
+            } catch (e) {
+              // we'll just bail and let it get logged as a string...
+            }
+          } else {
+            var error = new Error();
+            error.message = messageSplit[0];
+            error.stack = message;
+            messages = [error];
+          }
+        }
+      }
+    }
+    var scroll = false;
+    if (node) {
+      for (var i = 0; i < messages.length; i++) {
+        var message = messages[i];
+        if (Logger.isArray(message) || Logger.isObject(message)) {
+          var obj = "" ;
+          try {
+            obj = '<pre data-language="javascript">' + JSON.stringify(message, null, 2) + '</pre>';
+          } catch (error) {
+            obj = message + " (failed to convert) ";
+            // silently ignore, could be a circular object...
+          }
+          text = text + obj;
+        } else if (Logger.isError(message)) {
+          if ('message' in message) {
+            text = text + message['message'];
+          }
+          if ('stack' in message) {
+            postLog.push(function() {
+              var stackTrace = Logger.formatStackTraceString(message['stack']);
+              var logger = Logger;
+              if (context.name) {
+                logger = Logger.get(context['name']);
+              }
+              logger.info("Stack trace: ", stackTrace);
+            });
+          }
+        } else {
+          text = text + message;
+        }
+      }
+      if (context.name) {
+        text = '[<span class="green">' + context.name + '</span>] ' + text;
+      }
+      node.innerHTML = text;
+      node.className = context.level.name;
+      if (container) {
+        if (container.scrollHeight === 0) {
+          scroll = true;
+        }
+        if (panel.scrollTop > (panel.scrollHeight - container.scrollHeight - 200)) {
+          scroll = true;
+        }
+      }
+    }
+    function onAdd() {
+      if (panel && node) {
+        panel.appendChild(node);
+        if (panel.childNodes.length > parseInt(window['LogBuffer'])) {
+          panel.removeChild(panel.firstChild);
+        }
+        if (scroll) {
+          panel.scrollTop = panel.scrollHeight;
+        }
+      }
+      if (consoleLogger) {
+        consoleLogger(messages, context);
+      }
+      var interceptors = window['logInterceptors'];
+      for (var i = 0; i < interceptors.length; i++) {
+        interceptors[i](context.level.name, text);
+      }
+    }
+    onAdd();
+    postLog.forEach(function (func) { func(); });
```

Comment:
In fact, I think we can delete 120-250 also. Doesn't look like we're using any of it, and I'd rather not carry the code.